Common/Securities/MarketHoursDatabase.cs

```diff
@@ -91,6 +91,19 @@ public DateTimeZone GetDataTimeZone(string market, Symbol symbol, SecurityType s
             return GetEntry(market, stringSymbol, securityType).DataTimeZone;
         }
 
+        /// <summary>
+        /// Resets the market hours database, forcing a reload when reused.
+        /// Called in tests where multiple algorithms are run sequentially,
+        /// and we need to guarantee that every test starts with the same environment.
+        /// </summary>
+        public static void Reset()
+        {
+            lock (DataFolderMarketHoursDatabaseLock)
+            {
+                _dataFolderMarketHoursDatabase = null;
```

Comment:
The horrors of statics! not for now, but a later refactoring would be to remove these statics and the symbol properties statics and convert them into provides (similar to map/factor file) -- benefit would be could define a `GitHubMarketHoursDatabaseProvider` which fetches the file from github (same with symbol properties)!